docs/sanic/getting_started.rst

```diff
@@ -60,3 +60,21 @@ Open the address `http://0.0.0.0:8000 <http://0.0.0.0:8000>`_ in your web browse
 the message *Hello world!*.
 
 You now have a working Sanic server!
+
+5. Application registry
```

Comment:
Should we not at least mention the workflow needed for this to work properly? I mean, you should never run `get_app` prior to `Sanic("...")` ... Or, perhaps, make _only_ the `Sanic("...")` call, where if there's no instance called `foo`, it will create one, otherwise return that instance ? This would avoid the workflow problem I mentioned ...

Reply:
> When you instantiate a Sanic instance, that can be retrieved at a later time ...

We do. I was playing with the idea of allowing `get_app()` to create and return an instance. But, I like the explicit nature because it would avoid a potential bug. What if I created `Sanic("my_app")`, but accidentally ran `Sanic.get_app("myapp")`? I think allowing this to return a new instance would lead to hard to find bugs. Getting an explicit `SanicException` is pretty clear.

I then thought maybe we do this:

```python
try:
    return cls._app_registry[name]
except KeyError:
    if len(cls._app_registry) == 0:
        return cls(name)
    raise SanicException(f'Sanic app name "{name}" not found.')
```

It would be a nice fallback. Perhaps even throw in a warning. This one I am more open to, but still I think it suffers a problem because now the method does not always behave the same way. What about a flag (`strict=True`)? By the time we complicate this so much, it still doesn't get around the fact that this is a convenience, and should not be abused, or mistakenly implemented.

---

In the end, I think a simpler logic with exception leads to more explicit implementation with fewer pitfalls for the dev.

Reply:
@vltr any thoughts on the response to this? I'm okay with @ahopkins proposal for both adding the change in behavior along with the flag to manage it.

Reply:
@sjsadowski I had a conversation with @vltr offline. We decided that perhaps the clearest thing would be to raise an exception when it does not exist, but allow the method to force a new instance.

```python
    assert isinstance(
        Sanic.get_app("does-not-exist", force_create=True), Sanic
    )
```